xcodeproj/internal/xcode_targets.bzl

```diff
@@ -324,11 +328,22 @@ def _set_bazel_outputs_product(
     if build_mode != "bazel":
         return
 
-    path = xcode_target.outputs.product_path
-    if not path:
-        return
-
-    build_settings["BAZEL_OUTPUTS_PRODUCT"] = path
+    product_path = xcode_target.outputs.product_path
+    if product_path:
+        build_settings["BAZEL_OUTPUTS_PRODUCT"] = product_path
+    
+    dsym_files = xcode_target.outputs.dsym_files
+    if dsym_files:
+        dsym_paths = []
+        for file in dsym_files.to_list():
+            file_path = file.path
+            # dSYM files contain plist and DWARF.
+            if not file_path.endswith('Info.plist'):
+                # ../Product.dSYM/Contents/Resources/DWARF/Product
+                dsym_path =  "/".join(file_path.split('/')[:-4])
+                dsym_paths.append(dsym_path)
+        if dsym_paths:
+            build_settings["BAZEL_OUTPUTS_DSYM"] = "\n".join(dsym_paths)
```

Comment:
Per my other comment, each path should be wrapped in `"`, in case it has spaces in it.